File: Modify_the_Matrix.java
class Solution {
    public int[][] modifiedMatrix(int[][] matrix) {
        int m = matrix.length; // row
        int n = matrix[0].length; // column
        int[][] answer = new int[m][n];

        for (int r = 0; r < m; r++) {
            answer[r] = Arrays.copyOf(matrix[r], n);
        }

        for (int c = 0; c < n; c++) {
            int maxVal = Integer.MIN_VALUE;
            for (int r = 0; r < m; r++) {
                maxVal = Math.max(maxVal, matrix[r][c]);
            }
            for (int r = 0; r < m; r++) {
                if (matrix[r][c] == -1) {
                    answer[r][c] = maxVal;
                }
            }
        }

        return answer;        
    }
}

File: Modify_the_Matrix.py
class Solution:
    def modifiedMatrix(self, matrix: List[List[int]]) -> List[List[int]]:
        m = len(matrix) # row
        n = len(matrix[0]) # column
        answer = matrix.copy()

        for c in range(n):
            max_val = max(matrix[r][c] for r in range(m))
            for r in range(m):
                if matrix[r][c] == -1:
                    answer[r][c] = max_val
        
        return answer
